### PR TITLE
AP-2157 Modify CI to test install script on deploy

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?
Add a job to trigger the `agent-linux-install-script` pipeline, to fully test the install script.
Only triggered on deploy.

### Motivation

We want to have the install-script tested with the agent version we deploy, to be sure the agent can be installed properly
### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: .gitlab/install_script_testing.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    - source /root/.bashrc
+    - set +x
+    - export GITLAB_TOKEN=$(aws ssm get-parameter --region us-east-1 --name ci.datadog-agent.gitlab_pipelines_scheduler_token --with-decryption --query "Parameter.Value" --out text)
+    - export APT_URL=$DEB_TESTING_S3_BUCKET
+    - export YUM_URL=$RPM_TESTING_S3_BUCKET
+    - export APT_REPO_VERSION_AGENT6="pipeline-$CI_PIPELINE_ID-a6 6"
+    - export YUM_VERSION_PATH_AGENT6="testing/pipeline-$CI_PIPELINE_ID-a6/6"
+    - export APT_REPO_VERSION_AGENT7="pipeline-$CI_PIPELINE_ID-a7 7"
+    - export YUM_VERSION_PATH_AGENT7="testing/pipeline-$CI_PIPELINE_ID-a7/7"
+    - inv pipeline.trigger-child-pipeline --project-name "DataDog/agent-linux-install-script" --git-ref "feat/install-script-test-on-release" --variables "APT_URL,YUM_URL,APT_REPO_VERSION_AGENT6,APT_REPO_VERSION_AGENT7,YUM_VERSION_PATH_AGENT6,YUM_VERSION_PATH_AGENT7"
```
Comment (chouetz) [inline .gitlab/install_script_testing.yml:16]: Shouldn't we make this test mandatory with the `--follow` option (even if it increases the pipeline duration, on deploy it's maybe acceptable)

**Review by chouetz — APPROVED** [review]
If you confirm we don't want a mandatory test I'm ok

File: .gitlab/install_script_testing.yml (hunk 1 above)
Comment (KevinFairise2) [reply .gitlab/install_script_testing.yml:16]: You're right I think it's better if we use `--follow` so we can see in the pipeline if it failed

Comment (KevinFairise2) [reply .gitlab/install_script_testing.yml:16]: Actually default value for `follow` is true in the invoke task. For other jobs we usually set `--no-follow` to child pipeline we do not want to follow. So I can probably let it as is and the child pipeline is followed

Comment (chouetz) [reply .gitlab/install_script_testing.yml:16]: you're right, thanks!

**Pull request merged**